eMPRess_GUI_version1.py

```diff
@@ -0,0 +1,354 @@
+        recon_dup_label.grid(row=2, column=0, sticky="w")
+        recon_trans_label.grid(row=3, column=0, sticky="w")
+        recon_loss_label.grid(row=4, column=0, sticky="w")
+
+    def open_and_close_window_recon_space(self):
+        """
+        Opens a new window titled "View reconciliation space" when the checkbox is checked,
+        and closes the window when the checkbox is unchecked.
+        """
+        if self.recon_space_btn_var.get() == 1:
+            self.recon_space_window = tk.Toplevel(self.master)
+            self.recon_space_window.geometry("400x400")
+            self.recon_space_window.title("View reconciliation space")
+            Recon_space_window(self.recon_space_window)
+        if self.recon_space_btn_var.get() == 0:
+            if self.recon_space_window.winfo_exists() == 1:
+                self.recon_space_window.destroy()
+
+    def open_and_close_window_recons(self):
+        """
+        Opens a new window titled "View reconciliations" when the checkbox is checked,
+        and closes the window when the checkbox is unchecked.
+        """
+        if self.recons_btn_var.get() == 1:
+            self.recons_window = tk.Toplevel(self.master)
+            self.recons_window.geometry("400x400")
+            self.recons_window.title("View reconciliations")
+            Recon_window(self.recons_window)
+        if self.recons_btn_var.get() == 0:
+            if self.recons_window.winfo_exists() == 1:
+                self.recons_window.destroy()
+
+    def open_and_close_window_histogram(self):
+        """
+        Opens a new window titled "View p-value histogram" when the checkbox is checked,
+        and closes the window when the checkbox is unchecked.
+        """
+        if self.histogram_btn_var.get() == 1:
+            self.histogram_window = tk.Toplevel(self.master)
+            self.histogram_window.geometry("400x400")
+            self.histogram_window.title("View p-value histogram")
+            Histogram_window(self.histogram_window)
+        if self.histogram_btn_var.get() == 0:
+            if self.histogram_window.winfo_exists() == 1:
+                self.histogram_window.destroy()
+
+### View reconciliation space ###
+class Recon_space_window:
+    def __init__(self, master):
+        self.master = master        
+        self.frame = tk.Frame(master)
+        self.frame.pack(fill=tk.BOTH, expand=1)
+        self.frame.pack_propagate(False)
+
+### View reconciliations ###
+class Recon_window:
+    def __init__(self, master):
+        self.master = master        
+        self.frame = tk.Frame(master)
+        self.frame.pack(fill=tk.BOTH, expand=1)
+        self.frame.pack_propagate(False)
+
+### View p-value histogram ###
+class Histogram_window:
+    def __init__(self, master):
+        self.master = master        
+        self.frame = tk.Frame(master)
+        self.frame.pack(fill=tk.BOTH, expand=1)
+        self.frame.pack_propagate(False)
+
+def on_closing():
+   plt.close("all")
```

Comment:
It would be nice if you create an issue that explains why this line is necessary and close the issue, and then put the issue number or the link to the issue on this line. Alternatively, you could also write why you need to do this on this line.

Reply:
A comment was added for the on_closing function. It kills the matplotlib program and all other tkinter programs when the master window is closed.

Reply:
Could you create an issue for this or write a comment that explains what will happen if you remove this line? Something like `# If we don't run plt.close("all"), then python will not terminate after the gui exits`. Alternatively, write something like this as a comment in the file.